pylearn2/linear/cudnn2d.py

```diff
@@ -0,0 +1,203 @@
+import functools
+import numpy as np
+
+from theano.sandbox.cuda.dnn import GpuDnnConv, GpuDnnConvDesc
+from theano.sandbox.cuda.basic_ops import gpu_contiguous
+
+from pylearn2.packaged_dependencies.theano_linear.conv2d \
+    import Conv2d as OrigConv2D
+
+from pylearn2.linear.linear_transform import LinearTransform as P2LT
+from pylearn2.utils import sharedX
+from pylearn2.utils.rng import make_np_rng
+
+
+default_seed = [2012, 11, 6, 9]
+default_sparse_seed = [2012, 11, 6]
+
+
+class Cudnn2D(OrigConv2D):
+    """
+    Wrapper on the Theano Cudnn op.
+
+    Parameters
+    ----------
+    filters : Theano shared variable
+        4D-tensor of shape (out channels, in channels, rows, cols)
+    batch_size : int
+        The size of the input batches
+    input_space : Space
+        The Space of the input data
+    output_axes : tuple, optional
+        The requested output axes. If not specified `bc01` will be used.
```

Comment:
I don't think it matters much, as the output_axes are currently always provided explicitly.
It is true that dimshuffles are performed inplace with almost no overhead, and chained dimshuffles would even be fused by Theano. However, there is a slightly higher risk that a "gpu_contiguous" gets inserted between dimshuffles, which would force a copy.
I would keep bc01, but I would not bother making the same change in conv2d.Conv2D.